docs/tutorials/typescript/subscriptions/index.md

```diff
@@ -1,416 +1,417 @@
+In the code above, we define two Signals:
+
+1. `cancelSubscription`: A Signal with no input parameters, used to cancel the subscription.
+2. `updateBillingChargeAmount`: A Signal that takes a single parameter of type number, used to update the billing charge amount.
+
+However, just defining a Signal isn't enough. We also need to let the Workflow know what to do when it receives a Signal of that type. This is done with a Signal handler. Signal handlers are functions within the Workflow that listen for incoming Signals of a specified type and define how the Workflow should react.
 
-## Data Model
+In order to implement a Signal handler, you will need to use the [`setHandler`](https://typescript.temporal.io/api/namespaces/workflow/#sethandler) function provided by the TypeScript SDK API. This function associates each Signal with its corresponding handler logic.
 
-We are about to start charging for money and adding more settable properties like `BillingPeriod` and `BillingPeriodChargeAmount`.
-It's time to evolve the data model accordingly.
-We'll define a shared type:
+Within the `workflows.ts` file, implement the Signal handlers as shown below:
 
+<!--SNIPSTART subscription-ts-workflow-definition {"selectedLines": ["45-53"]}-->
+[src/workflows.ts](https://github.com/temporalio/subscription-workflow-project-template-typescript/blob/main/src/workflows.ts)
 ```ts
-export type Customer = {
-  email: string;
-  trialPeriod: number;
-  billingPeriod: number;
-  maxBillingPeriods: number;
-  initialBillingPeriodCharge: number;
-  id: string;
-};
+// ...
+  setHandler(cancelSubscription, () => {
+    subscriptionCancelled = true;
+  });
+  setHandler(updateBillingChargeAmount, (newAmount: number) => {
+    billingPeriodChargeAmount = newAmount;
+    log.info(
+      `Updating BillingPeriodChargeAmount to ${billingPeriodChargeAmount}`
+    );
+  });
 ```
+<!--SNIPEND-->
 
-And refactor our existing Workflow and Client code accordingly.
-We'll spare you the gory details - you can always [check our repo](https://github.com/temporalio/subscription-workflow-project-template-typescript) if you like, but there is no right answer here.
+1. When the `subscriptionWorkflow` Execution is running and it receives the `cancelSubscription` Signal, the corresponding handler function sets `subscriptionCancelled` to `true`. This effectively cancels the subscription.
+2. Similarly, when the `updateBillingChargeAmount` Signal is received, the handler updates `billingPeriodChargeAmount` to the new value provided by the Client. This mechanism allows the Workflow to dynamically respond to external events without interrupting its execution flow.
 
-## Write the Billing Process
+You have now learned what a Query and Signal is, as well as how to define and handle them in your Workflow Definition. In the next section, you will learn how to send a Signal and Query.
 
-We are now at Requirement 3:
+### Invoke a Signal from the Client
```

Comment:
One of the things I like about this version of the tutorial in Go, is that it contextualizes the use of the Client within an HTTP server: https://learn.temporal.io/tutorials/go/subscriptions/#build-the-web-server

Possible to level this section up to something like that?

Reply:
Thanks for the suggestion! Introducing HTTP server setup and specifics would shift the focus away from the core topic I think in this tutorial. With all due respect, I think I will keep this tutorial simplified for now - thanks!